### PR TITLE
stats branch. JBD scripts.

### PR DESCRIPTION
**Pull request by thomasarsouze**
Would be nice, starting from JBD scripts, to adapt to the workflow in branch `openalea_docker_images_gitlabCIRAD`.

### CODE REVIEW

File: stats_openalea/vplants.def [hunk 1]
```diff
@@ -0,0 +1,262 @@
+Bootstrap: docker
+From: ubuntu:16.04
```
Comment (pradal) [inline stats_openalea/vplants.def:2]: 18.04 may make the trick

File: stats_openalea/vplants.def [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+singularity run --app console -e vplants_1_0.simg
+If you need to run x11-based applications, first set DISPLAY as in host
+
+%labels
+AUTHOR Jean-Baptiste Durand
+VERSION="1.0"
+VCS-URL="https://gricad-gitlab.univ-grenoble-alpes.fr/durandje/vplants.bak.git"
+VCS-REF="6654a6859651640f9b0e4a12d2efa28683e2313e"
+BUILD.CMD="sudo singularity build vplants_1_0.simg Docker-Singularity/vplants.def"
+
```
Comment (pradal) [inline stats_openalea/vplants.def:23]: vplants is now on github.com/VirtualPlants/AML

File: stats_openalea/vplants.def [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+# R repository mirror to install extra packages
+export R_REPOS="https://stat.ethz.ch"
+
+# git clone https://gricad-gitlab.univ-grenoble-alpes.fr/durandje/vplants.bak.git
+
+# files to copy inside the image
+%files
+./openalea-stable/ /mnt/devlp/openalea-stable/
+./vplants-stable-with-aml/ /mnt/devlp/vplants-stable-with-aml/
+
```
Comment (pradal) [inline stats_openalea/vplants.def:37]: Depends on openalea sconsx and deploy

File: stats_openalea/vplants.def [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+# Change mirror for ubuntu packages
+# sed -i 's|http://archive.ubuntu.com/|http://fr.archive.ubuntu.com/|g' /etc/apt/sources.list
+
+apt update && apt install -y gedit && \
+    apt install -y git
+    
+# Install packages for openalea / vplants compilation
+apt install -y freeglut3
+apt install -y freeglut3-dev
+apt install -y libboost-python1.58.0 libboost1.58-dev libboost-python1.58-dev
```
Comment (pradal) [inline stats_openalea/vplants.def:53]: Update version of boost

File: stats_openalea/vplants.def [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+# Install packages required at execution
+
+# Solve issue ImportError: cannot import name 'main'
+# in pip 
+
+hash pip
+
+# pip install --upgrade pip==19.3.1
+
+python2 -m pip install --upgrade pip==19.3.1
```
Comment (pradal) [inline stats_openalea/vplants.def:73]: conda install?

File: stats_openalea/vplants.def [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+# Compile openalea packages
+cd /mnt/devlp/openalea-stable/openalea/
+python multisetup.py install
+alea_config
+
+# Compile vplants packages
+cd /mnt/devlp/vplants-stable-with-aml/tool
+python setup.py install
+alea_config
+cd ../PlantGL
```
Comment (pradal) [inline stats_openalea/vplants.def:126]: PlantGL is not needed

**Review by pradal — APPROVED** [review]

**Comment by thomasarsouze** [comment]
Thanks @pradal for the review. Will test the generation of docker image according to your changes with @jbdurand.

**Comment by jbdurand** [comment]
I just updated the dockerfile in stats branch. @pradal it should be sufficient now to indicate source repositories to AML and sequence_analysis.

**Comment by pradal** [comment]
Thanks @jbdurand .
Things have changed a lot:
- there is no alea-config anymore
- there is no multisetup.py

What you need is:
- install openalea.deploy 
    `conda install -c conda-forge openalea.deploy`
- install openalea.sconsx 
    `conda install openalea.sconsx -c openalea3 -c conda-forge`

Then, install for github:
-  AML
    `git clone https://github.com/VirtualPlants/AML`
- StructureAnalysis
```
git clone https://github.com/openalea/StructureAnalysis.git
git checkout aml
```

**Comment by jbdurand** [comment]
Thanks @pradal ; I added the lines in the docker recipe. 
However, when running 
```
cd StructureAnalysis/tool && \
    python setup.py install
```
I get 
```
Error
 File "/root/anaconda3/envs/aml/lib/python3.11/configparser.py", line 443, in _interpolate_some
   raise InterpolationSyntaxError(
configparser.InterpolationSyntaxError: '%' must be followed by '%' or '(', found: '%{name} = %{version}'
```
If I try 
```
cd StructureAnalysis/tool && \
    python2.7 setup.py install
```
then I get 
```
Error
Traceback (most recent call last):
 File "setup.py", line 9, in <module>
   from openalea.deploy.metainfo import read_metainfo
ImportError: No module named openalea.deploy.metainfo
```
Basically I cannot figure out how to compile tool, which obviously requires python2.7, with a python3 conda version of openalea.deploy?

**Comment by pradal** [comment]
Yes. Things have changed a little bit.
No metainfo and we need to remove setup.cfg

**Comment by jbdurand** [comment]
So as I understand, the right command is 
`python2.7 setup.py install`
In this case, what do we have to change in setup.py? 
Just removing 
```
from openalea.deploy.metainfo import read_metainfo
metadata = read_metainfo('metainfo.ini', verbose=True)
for key,value in list(metadata.items()):
   exec("%s = '%s'" % (key, value))

```
would cause
```
Traceback (most recent call last):
  File "setup.py", line 29, in <module>
    setup(name=name,
NameError: name 'name' is not defined

```

**Comment by pradal** [comment]
The content of the metainfo.ini:
name, version, ...

**Comment by jbdurand** [comment]
Could you please modify the file (setup.py) on the repository and create the metainfo accordingly, at least on "tool"?

**Comment by pradal** [comment]
Sure. I will do that next week.
Open an issue and assign it to me. THs

**Comment by jbdurand** [comment]
Thanks; I opened an issue but it seems I cannot assign it. I checked how to assign at 
[https://docs.github.com/en/issues/tracking-your-work-with-issues/assigning-issues-and-pull-requests-to-other-github-users](https://docs.github.com/en/issues/tracking-your-work-with-issues/assigning-issues-and-pull-requests-to-other-github-users)
but I cannot see any gears icon or action related to the issue; probably I do not have rights to assign.

**Pull request closed without merging**